cmd/armada/main.go

```diff
@@ -63,19 +82,22 @@ func main() {
 	defer shutdownGateway()
 
 	// start HTTP server
+	// TODO: Run in errgroup
 	shutdownHttpServer := common.ServeHttp(config.HttpPort, mux)
 	defer shutdownHttpServer()
 
-	shutdown, wg := armada.Serve(&config, healthChecks)
-	go func() {
-		<-stopSignal
+	// Start Armada server
+	g.Go(func() error {
+		return armada.Serve(ctx, &config, healthChecks)
+	})
 
-		// TODO Can't be sure this will run, since there are calls to panic/os.Exit in sub-calls
-		shutdown()
+	// Assume the server is ready if there are no errors within 10 seconds.
+	go func() {
+		time.Sleep(10 * time.Second)
```

Comment:
We didn't have one before because we assumed that services were ready immediately after being started. This is more conservative. I agree, but I'm not yet sure how to implement it.